src/explore-education-statistics-common/src/modules/table-tool/components/TimePeriodDataTable.tsx

```diff
@@ -56,11 +50,54 @@ function getExcludedFilters(
     ...subjectMeta.indicators,
   ].map(filter => filter.id);
 
-  return subjectMetaFilters.filter(
-    subjectMetaFilter => !tableHeaderFilters.includes(subjectMetaFilter),
+  return new Set(
+    subjectMetaFilters.filter(
+      subjectMetaFilter => !tableHeaderFilters.includes(subjectMetaFilter),
+    ),
   );
 }
 
+/**
+ * Determines whether any rows or columns are excluded from the table because they have no data.
+ * For filters, indicators and locations:
+ *  - when there's no data, these aren't included in subjectMeta.
+ *  - compare subjectMeta with the query to find the excluded ones.
+ *  For timePeriod:
+ *  - when there's no data, these are in subjectMeta but aren't in tableHeadersConfig.
+ *  - compare subjectMeta with tableHeadersConfig to find the excluded ones.
+ *  - the query isn't useful here as just has the start and end timePeriods.
+ */
+const excludesRowsOrColumnsWithNoData = (
```

Comment:
'Excludes' might be the wrong terminology here as we can mix this up with the `getExcludedFilters` function. Perhaps `hasMissingRowsOrColumns` might be sufficient?